lib/Net/Braintree/Transaction.pm

```diff
@@ -50,9 +51,12 @@ sub void {
 }
 
 sub refund {
+  # Follow-on credit
   my ($class, $id, $amount) = @_;
-  my $params = {};
-  $params->{'amount'} = $amount if $amount;
+  my $params = ref ($amount) eq 'HASH'
```

Comment:
Does this behave differently when `$amount` is falsy (for example, when it's `0`)? We don't want to introduce a breaking change.